tech/firewall-rule-format.md

```diff
@@ -575,15 +600,487 @@ For the record, the reason why the protocol is `*` in Azure is that
+    }
+
+Although Azure portal does not allow `0.0.0.0/0` as a CIDR range for
+`source_address_prefix`, we should still check for it to be extra safe.
+It may be possible that there are other means (such as a direct REST API
+call) to set `source_address_prefix` to `0.0.0.0/0`.
+
+TODO: Check if we have any data that shows `source_address_prefix` set
+to `0.0.0.0/0`.
+
+
+#### `source_address_prefixes` set to multiple CIDRs
+
+Here is yet another way to allow the entire Internet to allow access to
+port 22. This security rule was created by choosing *IP Addresses* from
+the *Source* dropdown and setting *Source IP addresses/CIDR ranges* as
+`0.0.0.0/1,128.0.0.0/1`.
+
+    {
+      "id": "/subscriptions/a8daa361-b951-46b5-8e9d-b375f2ffe9fd/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/vm1-nsg/securityRules/Port_22",
+      "protocol": "*",
+      "source_port_range": "*",
+      "destination_port_range": "22",
+      "source_address_prefixes": [
+        "0.0.0.0/1",
+        "128.0.0.0/1"
+      ],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [],
+      "access": "Allow",
+      "priority": 100,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_22",
+      "etag": "W/\"7a502bb3-4f77-4c23-a5fa-1fda055b021d\""
+    }
+
+To take multiple CIDRs into account and finding whether their union is
+`0.0.0.0/0` should perhaps be outside the scope of MVP. Detecting this
+kind of corner cases (which could be exploited as clever workarounds to
+avoid detection) should perhaps be picked up as enhancements later.
+
+Until then, perhaps to be extra safe we should look for `0.0.0.0/0` as
+one of the entries in the `source_address_prefixes` list.
+
+
+#### `source_address_prefix` set to `Internet`
+
+Yet another way to expose a port to the Internet is to use a service
+tag. This security rule was created by choosing *Service Tag* from the
+*Source* dropdown menu and choosing *Internet* from the *Source service
+tag* dropdown menu.
+
+    {
+      "id": "/subscriptions/a8daa361-b951-46b5-8e9d-b375f2ffe9fd/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/vm1-nsg/securityRules/Port_22",
+      "protocol": "*",
+      "source_port_range": "*",
+      "destination_port_range": "22",
+      "source_address_prefix": "Internet",
+      "source_address_prefixes": [],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [],
+      "access": "Allow",
+      "priority": 100,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_22",
+      "etag": "W/\"a9ceecf4-eb4a-4fb8-b961-4cd9ab246e70\""
+    }
+
+
+### Any Source: GCP
+
+GCP contains this firewall rule to allow connectivity port 22 from
```

Comment:
"GCP contains this firewall rule to allow connectivity **to** port 22 from"

Reply:
Addressed in [line 781](https://github.com/cloudmarker/guides/pull/5/files#diff-112a678346155cbc8647b65f47882715R781).